src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -99,6 +99,17 @@ public ILibraryCatalog GetCatalog()
             return LibraryInstallationResult.FromSuccess(desiredState);
         }
 
+        /// <summary>
+        /// No-op for FileSystemProvider
+        /// </summary>
+        /// <param name="desiredState"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
```

Comment:
We need to implement this for filesystem as well. Restore will fail if we don't specify the files collection as well. Null ref exception. We need to be able to populate the Files collection based on the "library" specified. 
If path is to a file, add that file to the Files collection. If path is to a folder, add all the files from that folder.

Reply:
There is another bug tracking this work item